libnymea-modbus/tools/generate-connection.py

```diff
@@ -482,48 +482,127 @@ def writeRtuSourceFile():
 # Main
 ############################################################################################
 
+logger = logging.getLogger('modbus-tools')
+logger.setLevel(logging.INFO)
+ch = logging.StreamHandler(sys.stdout)
+ch.setLevel(logging.INFO)
+formatter = logging.Formatter('%(name)s: %(message)s')
+ch.setFormatter(formatter)
+logger.addHandler(ch)
+
 parser = argparse.ArgumentParser(description='Generate modbus tcp connection class from JSON register definitions file.')
 parser.add_argument('-j', '--json', metavar='<file>', help='The JSON file containing the register definitions.')
 parser.add_argument('-o', '--output-directory', metavar='<directory>', help='The output directory for the resulting class.')
-parser.add_argument('-c', '--class-name', metavar='<name>', help='The name of the resulting class.')
+parser.add_argument('-v', '--verbose', dest='verboseOutput', action='store_true', help='More verbose output.')
 args = parser.parse_args()
 
 registerJson = loadJsonFile(args.json)
 scriptPath = os.path.dirname(os.path.realpath(sys.argv[0]))
 outputDirectory = os.path.realpath(args.output_directory)
-className = args.class_name
 
-headerFileName = className.lower() + '.h'
-sourceFileName = className.lower() + '.cpp'
+if not os.path.exists(outputDirectory):
+    logger.debug("Output directory does not exist. Creating directory %s", outputDirectory)
+    os.makedirs(outputDirectory)
+
+if args.verboseOutput:
+    logger.setLevel(logging.DEBUG)
+    ch.setLevel(logging.DEBUG)
 
-headerFilePath = os.path.join(outputDirectory, headerFileName)
-sourceFilePath = os.path.join(outputDirectory, sourceFileName)
+logger.debug("Verbose output enabled")
 
-print('Scrip path: %s' % scriptPath)
-print('Output directory: %s' % outputDirectory)
-print('Class name: %s' % className)
-print('Header file: %s' % headerFileName)
-print('Source file: %s' % sourceFileName)
-print('Header file path: %s' % headerFilePath)
-print('Source file path: %s' % sourceFilePath)
+if not 'className' in registerJson:
+    logger.warning('Classname missing. Please specify the classname in the json file or pass it to the generatori using -c .')
+    exit(1)
+
+classNamePrefix = registerJson['className']
 
 endianness = 'BigEndian'
 if 'endianness' in registerJson:
     endianness = registerJson['endianness']
 
+logger.debug('Scrip path: %s' % scriptPath)
+logger.debug('Output directory: %s' % outputDirectory)
+logger.debug('Class name prefix: %s' % classNamePrefix)
+logger.debug('Endianness: %s' % endianness)
+
 protocol = 'TCP'
 if 'protocol' in registerJson:
     protocol = registerJson['protocol']
 
 if 'blocks' in registerJson:
     validateBlocks(registerJson['blocks'])
 
+# Create classes depending on the protocol
+writeTcp = True
+writeRtu = False
+
 if protocol == 'TCP':
+    writeTcp = True
+    writeRtu = False
+elif protocol == 'RTU':
+    writeTcp = False
+    writeRtu = True
+else:
+    # Any other value generates both classes
+    writeTcp = True
+    writeRtu = True
```

Comment:
would probably have used
```
writeTcp = protocol in ["TCP", "BOTH"]
writeRtu = protocol in ["RTU", "BOTH"]
```
and not generate both files if the user makes a typo in the json.

Reply:
Updated logic: e1d3593dbceca21ede514d018729a41d0e58c6e0